plugins/WindowManager/Window.cpp

```diff
@@ -184,7 +184,13 @@ void Window::setSurface(unityapi::MirSurfaceInterface *surface)
             setSurface(nullptr);
         });
 
+        // Surface should never be focused at this point!
+        Q_ASSERT(!m_surface->focused());
```

Comment:
If this assertion fails, Unity crashes, correct?

Reply:
If debug is enabled (which it should be), then yes, it does a `qFatal()` which is of course, fatal.

Reply:
Only if debug is enabled (as @dobey says). 

but else then debug mode, if this happens in production mode we should let the code go, since it will run updateFocused later where our state will be updated.

Reply:
If the idea is we should continue on here, but we want some logging, then I think it'd be better to use a `qCritical()` inside an `if` block.